src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift

```diff
@@ -321,8 +326,11 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
 					return
 				}
 				switch testResult {
-				case .negative, .positive, .invalid:
+				case .positive, .negative, .invalid:
 					self.store.testResultReceivedTimeStamp = Int64(Date().timeIntervalSince1970)
+					self.store.testResultDate = Date()
+					self.updateStoreWithHoursSinceHighRiskWarningAtTestRegistration()
+					self.updateStoreWithDaysSinceMostRecentDateAtRiskLevelAtTestRegistration()
 					completeWith(.success(testResult))
```

Comment:
again Please add relevant tests to the ExposureSubmissionServiceTests, you can check
  `func testGetTestResult_TestRetrievalsucceed_TestMetadataCreated()` as an example to test the getTestResult() function

Reply:
Created a separate Task for Unit Testing. Please refer to 5221